coredocument/coredocument.go

```diff
@@ -283,38 +335,31 @@ func GetTypeURL(coreDocument *coredocumentpb.CoreDocument) (string, error) {
 
 // CreateProofs util function that takes document data tree, coreDocument and a list fo fields and generates proofs
 func CreateProofs(dataTree *proofs.DocumentTree, coreDoc *coredocumentpb.CoreDocument, fields []string) (proofs []*proofspb.Proof, err error) {
-	dataRootHashes, err := getDataProofHashes(coreDoc)
-	if err != nil {
-		return nil, errors.New("createProofs error %v", err)
-	}
-
-	signingRootHashes, err := getSigningProofHashes(coreDoc)
+	signingRootProofHashes, err := getSigningRootProofHashes(coreDoc)
 	if err != nil {
 		return nil, errors.New("createProofs error %v", err)
 	}
 
-	cdtree, err := GetDocumentSigningTree(coreDoc)
+	cdtree, err := GetDocumentCDTree(coreDoc)
 	if err != nil {
 		return nil, errors.New("createProofs error %v", err)
 	}
 
 	// We support fields that belong to different document trees, as we do not prepend a tree prefix to the field, the approach
 	// is to try in both trees to find the field and create the proof accordingly
 	for _, field := range fields {
-		rootHashes := dataRootHashes
 		proof, err := dataTree.CreateProof(field)
 		if err != nil {
 			if strings.Contains(err.Error(), "No such field") {
 				proof, err = cdtree.CreateProof(field)
 				if err != nil {
 					return nil, errors.New("createProofs error %v", err)
 				}
-				rootHashes = signingRootHashes
 			} else {
 				return nil, errors.New("createProofs error %v", err)
 			}
 		}
-		proof.SortedHashes = append(proof.SortedHashes, rootHashes...)
+		proof.SortedHashes = append(proof.SortedHashes, signingRootProofHashes...)
```

Comment:
Need to append the proof for sibling hash for either the `data_root` or `coredoc_root`